### PR TITLE
Download file reference directly from source

### PR DESCRIPTION
**Pull request by hmusum**
When getting a 'filedistribution.setFileReferencesToDownload' RPC request, use the fact that the client is the one that has the file to download from that same source instead of potentially trying some other config server that might not have it.

See https://github.com/vespa-engine/vespa/issues/32153

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/rpc/RpcServer.java [hunk 1]
```diff
@@ -574,14 +574,19 @@ private void serveFile(Request request) {
...
+                    var client = req.target();
+                    var peerSpec = client.peerSpec();
                     Stream.of(fileReferenceStrings)
                             .map(FileReference::new)
-                            .forEach(fileReference -> downloader.downloadIfNeeded(
-                                    new FileReferenceDownload(fileReference,
-                                                              req.target().toString(),
-                                                              false /* downloadFromOtherSourceIfNotFound */)));
+                            .forEach(fileReference -> {
+                                log.log(FINE, "Downloading file reference " + fileReference.value() + " from " + peerSpec);
```
Comment (baldersheim) [inline configserver/src/main/java/com/yahoo/vespa/config/server/rpc/RpcServer.java:587]: I would log after downloadFromSource and include the return value in the log message.

**Review by baldersheim — DISMISSED** [review]
Looks good, just one comment to add some info in log message.

File: configserver/src/main/java/com/yahoo/vespa/config/server/rpc/RpcServer.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/rpc/RpcServer.java]: Yes, good idea, done

**Comment by hmusum** [comment]
@baldersheim please take a look again

**Review by baldersheim — APPROVED** [review]

**Pull request merged**